### PR TITLE
Result type refactor & auth middleware

### PR DESCRIPTION
**Pull request by seanim0920**

### CODE REVIEW

File: APILambda/user/updateUserProfile.ts [hunk 1]
```diff
@@ -47,27 +42,24 @@ const updateProfile = (
...
-    ? checkForUserWithHandle(conn, request.handle)
-    : success(undefined)
-  return getProfileSettings(conn, userId)
-    .flatMapSuccess((settings) => handleCheck.withSuccess(settings))
-    .flatMapSuccess((settings) => {
-      const handle = request.handle?.rawValue ?? settings.handle
-      const profile = { ...settings, ...request, handle }
-      return overwriteProfile(conn, userId, profile)
+) =>
+  checkForUserWithHandle(conn, request.handle)
```
Comment (seanim0920) [inline APILambda/user/updateUserProfile.ts]: Potentially add a method to merge results

File: APILambda/user/generateUserHandle.ts [hunk 2]
```diff
@@ -9,30 +9,26 @@ const generateNumericHash = (input: string) => {
...
+    .flatMapSuccess(() => success(potentialUsername))
+}
+
+export const generateUniqueUsername = (
   conn: SQLExecutable,
   name: string,
-  maxRetries = 3
-): Promise<string> => {
+  retries = 3
+) => {
```
Comment (seanim0920) [inline APILambda/user/generateUserHandle.ts:28]: try wrapping original method in a promiseresult

**Review by suryatiru — DISMISSED** [review]

File: APILambda/events/getChatToken.ts [hunk 3]
```diff
@@ -75,59 +57,27 @@ export const determineChatPermissions = (
...
-
-  const permissions = determineChatPermissions(
-    result.value.hostId,
-    result.value.endTimestamp,
-    userId,
-    eventId
+) =>
+  conn.transaction((tx) =>
+    getEventById(tx, eventId)
+      .flatMapSuccess(event => isUserInEvent(tx, userId, eventId).mapSuccess(() => event))
```
Comment (seanim0920) [inline APILambda/events/getChatToken.ts:66]: withSuccess

File: APILambda/user/createUserProfile.ts [hunk 4]
```diff
@@ -1,40 +1,101 @@
...
+          Name: "profile_created",
+          Value: "true"
+        }
+      ]
+    }
+
+  return promiseResult(success(cognito.adminUpdateUserAttributes(verifyEmailParams).promise())).withFailure("cannot-set-cognito-attribute" as const)
+}
+
+export const createUserProfileRouter = (
```
Comment (seanim0920) [inline APILambda/user/createUserProfile.ts:85]: Make type for each router and combine for whole app

File: GeocodingLambda/index.ts [hunk 5]
```diff
@@ -18,21 +20,14 @@ interface LocationSearchRequest extends Retryable {
...
+  Result<"placemark-successfully-inserted", "placemark-already-exists">
+>(
+  async (event: LocationSearchRequest) =>
+    checkExistingPlacemarkInDB({
+      latitude: parseFloat(event.location.latitude.toFixed(10)),
+      longitude: parseFloat(event.location.longitude.toFixed(10))
+    })
+      .flatMapSuccess(async () => success(await SearchClosestAddressToCoordinates(event.location)))
+      .flatMapSuccess((placemark) => addPlacemarkToDB(placemark))
+      .mapSuccess(() => "placemark-successfully-inserted" as const)
```
Comment (seanim0920) [inline GeocodingLambda/index.ts:32]: withsuccess

**Pull request merged**